pkg/aws/imds.go

```diff
@@ -19,6 +19,18 @@ func NewIMDSClient(config aws.Config) *imds.Client {
 	return imds.NewFromConfig(config)
 }
 
+// BuildIMDS builds or overrides the imds client in the Client with default aws config.
+func (c *Client) BuildIMDS(ctx context.Context) error {
```

Comment:
Is this required to make the `*Client` valid?

Reply:
yes, not sure if im following what you mean here